malformed/test_malformed_crlfs.py

```diff
@@ -0,0 +1,434 @@
+        has_resp = deproxy_cl.wait_for_response(timeout=5)
+        self.assertTrue(has_resp, "Response not received")
+        status = int(deproxy_cl.last_response.status)
+        self.assertTrue(status == expect_status, "Wrong status: "+str(status)+", expected: "+str(expect_status))
+        if expect is None:
+            self.assertTrue(deproxy_srv.last_request is None, "Request was unexpectedly sent to backend")
+        else:
+            #print('-----')
+            #print(deproxy_srv.last_request.original_data)
+            #print('-----')
+            #print(expect)
+            #print('-----')
+            if expect:
+                self.assertTrue(self.compare_head(deproxy_srv.last_request.original_data, expect), "Request sent to backend differs from expected one")
+
+    def extract_head(self, a):
+        p = a.find("Host:")
+        return a[0:p]
+
+    def compare_head(self, a, b):
+        a = self.extract_head(a)
+        b = self.extract_head(b)
+        return a == b
+
+    def test_no_crlf(self):
+    	# Test normal request
+    	#
+        request = 'GET / HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = 'GET / HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        self.common_check(request, 200, expect)
+
+    def test_no_crlf_pipeline(self):
+    	# Test 2 normal requests in pipeline
+    	#
+        request = 'GET /aaa HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n' \
+                  'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = 'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        self.common_check(request, 200, expect)
+
+    def test_single_crlf(self):
+    	# Test single CRLF before request
+    	# Request should be passed to backed with stripped CRLF
+    	# Proxy should return positive response
+    	#
+        request = \
+                  '\r\n' \
+                  'GET / HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = 'GET / HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        self.common_check(request, 200, expect)
+
+    def test_single_crlf_pipeline(self):
+    	# Test single CRLF before 2nd request in a pipeline
+    	# Request should be passed to backed with stripped CRLF
+    	# Proxy should return positive response
+    	#
+        request = \
+                  'GET /aaa HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n' \
+                  '\r\n' \
+                  'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = 'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        self.common_check(request, 200, expect)
```

Comment:
We need to verify that the both requests produced 200 status responses. I guess in this check you verify only the first status.

Reply:
@krizhanovsky 
I supposed, in opposite, that only the second status is checked there. I'll check it.
I do not think that we are to verify both statuses.

Reply:
agree